### PR TITLE
refactor(layout): Update CSS naming to match React implementation

### PR DESCRIPTION
**Pull request by jamesfan**
This PR updates the Layout (formerly known as Grid) CSS styles so they match the React component. 

|                  | Visual | Code |
|------------------|--------|------|
| Breaking Changes | No    | Yes   |

### Code Changes
- `lib/grid-system.scss` moved to `lib/layout.scss`
- `$wdc-grid-system-breakpoints` > `$wdc-layout-breakpoints`
- `$wdc-grid-system-columns` > `$wdc-layout-columns`

### Squash and Merge Message Proposal
refactor(layout): Update CSS styles to match React implementation (#168)

### CODE REVIEW

**Comment by jamesfan** [comment]
Here's what I changed:
- [x] Rename `grid-system.scss` to `layout.scss`
- [x] Rename `Grid` to `Layout` in stories
- [x] Rename `Grid` to `Layout` in README
- [x] Rename `gutter` to `spacing` in Sass variables
- [x] Rename `gutter` to `spacing` in README

I don't believe this is considered a breaking change since the module was already renamed before (though I did have to update the installation instructions in the `README`), but please correct me if I'm mistaken.

File: modules/layout/css/stories.tsx [hunk 1]
```diff
@@ -5,9 +5,9 @@ import README from './README.md';
 import './index.scss';
 import './stories.scss';
 
-storiesOf('CSS/Grid System', module)
+storiesOf('CSS/Layout', module)
   .addDecorator(withReadme(README))
-  .add('Responsive Grids', () => (
+  .add('Responsive Layout', () => (
     <div className="story story-grid">
```
Comment (jamesfan) [inline modules/layout/css/stories.tsx]: @sahlhoff btw, is `story-grid` needed? I don't see it referenced anywhere else, but I wanted to double-check with you before removing it.

Comment (sahlhoff) [reply modules/layout/css/stories.tsx]: @jamesfan good to remove 👍

File: modules/layout/css/lib/layout.scss [hunk 2]
```diff
@@ -9,9 +9,9 @@
 @each $breakpoint in $wdc-grid-system-breakpoints {
```
Comment (sahlhoff) [inline modules/layout/css/lib/layout.scss]: @jamesfan We should probably update all references of `grid-system` to `layout`

**Comment by jamesfan** [comment]
Pushed out changes addressing Chad's feedback above:
- [x] Remove `story-grid` class
- [x] Rename references of `grid(-system)` to `layout`

Let me know if you see any other issues. Otherwise, everything should be good to go.

**Comment by jamesfan** [comment]
@anicholls Got it, thanks for updating the description. I never added `BREAKING CHANGE` to my commit messages, though. Should I do something about that?

**Comment by anicholls** [comment]
Generally, yes - but since all of these commits went into `css-update` which contains a bunch of `BREAKING CHANGE` commits already, that will get caught when `css-update` is merged into `master`

**Pull request merged**